### PR TITLE
feat: added support for merchant view android

### PR DESCRIPTION
**Pull request by Georgepadannamackal**
Adding support for getMerchantView for react native plugin

### CODE REVIEW

File: android/src/main/java/in/juspay/hypersdkreact/HyperSdkReactModule.java [hunk 1]
```diff
@@ -50,6 +57,9 @@ public class HyperSdkReactModule extends ReactContextBaseJavaModule implements A
     static final String NAME = "HyperSdkReact";
     private static final String HYPER_EVENT = "HyperEvent";
 
+    @Nullable
+    private ReactInstanceManager rim;
```
Comment (harshgarg18) [inline android/src/main/java/in/juspay/hypersdkreact/HyperSdkReactModule.java:61]: Can we use proper names for this and rrv?

File: android/src/main/java/in/juspay/hypersdkreact/HyperSdkReactModule.java [hunk 2]
```diff
@@ -248,6 +261,31 @@ public void onEvent(JSONObject data, JuspayResponseHandler handler) {
...
+                    @Nullable
+                    @Override
+                    public View getMerchantView(ViewGroup viewGroup, MerchantViewType merchantViewType) {
+                        if (rim == null) {
+                            return super.getMerchantView(viewGroup, merchantViewType);
+                        } else {
+                            ReactRootView rrv = new ReactRootView(getCurrentActivity());
+                            switch (merchantViewType) {
+                                case HEADER:
+                                    rrv.startReactApplication(rim, "JuspayHeader");
```
Comment (harshgarg18) [inline android/src/main/java/in/juspay/hypersdkreact/HyperSdkReactModule.java:274]: As these constants are used in native code, we can expose them from here itself to js/ts to merchant

File: src/index.tsx [hunk 3]
```diff
@@ -28,6 +28,9 @@ console.log('HyperSDKReactVersion', HyperSdkReact);
 
 if (Platform.OS === 'android') {
   HyperSdkReact.updateBaseViewController = () => {};
+  HyperSdkReact.updateMerchantViewHeight = () => {
```
Comment (harshgarg18) [inline src/index.tsx:31]: can we use the same signature here as we have for iOS? 

```suggestion
  HyperSdkReact.updateMerchantViewHeight = (tag: string, height: number) => {
```

File: ios/HyperSdkReact.mm
Comment (harshgarg18) [inline ios/HyperSdkReact.mm:1]: @balaganesh-juspay - could you please review this?

**Review by harshgarg18 — CHANGES_REQUESTED** [review]

File: example/index.js [hunk 4]
```diff
@@ -1,5 +1,34 @@
...
 import { name as appName } from './app.json';
+import { NativeModules, Platform } from 'react-native';
+
+const LINKING_ERROR =
+  `The package 'HyperAPIUtils' doesn't seem to be linked. Make sure: \n\n` +
+  Platform.select({ ios: "- You have run 'pod install'\n", default: '' }) +
+  '- You rebuilt the app after installing the package\n' +
+  '- You are not using Expo Go\n';
+
+const HyperSdkReact = NativeModules.HyperSdkReact
```
Comment (harshgarg18) [inline example/index.js:14]: this is the incorrect way of importing HyperSdkReact in example.

Can you import via 'hyper-sdk-react' package

**Review by harshgarg18 — CHANGES_REQUESTED** [review]

File: src/index.tsx (hunk 3 above)
Comment (harshgarg18) [reply src/index.tsx:31]: this change is still pending

**Comment by harshgarg18** [comment]
there are conflicts in the PR. Can you rebase and resolve them?

**Comment by abhijeetvermayash** [comment]
POT 

https://github.com/juspay/hyper-sdk-react/assets/42650474/a13d88db-977c-4256-b0a4-b2bf469f2621

File: android/src/main/java/in/juspay/hypersdkreact/HyperSdkReactModule.java [hunk 5]
```diff
@@ -153,6 +163,8 @@ public static Set<Integer> getPermissionRequestCodes() {
     public Map<String, Object> getConstants() {
         return new HashMap<String, Object>() {{
             put(HYPER_EVENT, HYPER_EVENT);
+            put(MerchantViewConstants.JUSPAY_HEADER, MerchantViewConstants.JUSPAY_HEADER);
+            put(MerchantViewConstants.JUSPAY_HEADER_ATTACHED, MerchantViewConstants.JUSPAY_HEADER_ATTACHED);
         }};
```
Comment (harshgarg18) [inline android/src/main/java/in/juspay/hypersdkreact/HyperSdkReactModule.java:168]: We should also send FOOTER AND FOOTER_ATTACHED

File: android/src/main/java/in/juspay/hypersdkreact/HyperSdkReactModule.java [hunk 6]
```diff
@@ -248,6 +263,31 @@ public void onEvent(JSONObject data, JuspayResponseHandler handler) {
                         }
                         sendEventToJS(data);
                     }
+
+                    @Nullable
+                    @Override
+                    public View getMerchantView(ViewGroup viewGroup, MerchantViewType merchantViewType) {
+                        if (reactInstanceManager == null) {
```
Comment (harshgarg18) [inline android/src/main/java/in/juspay/hypersdkreact/HyperSdkReactModule.java:270]: you need to add null check for activity too

File: example/android/app/src/main/assets/index.android.bundle
Comment (harshgarg18) [inline example/android/app/src/main/assets/index.android.bundle:1]: don't commit this file

File: example/android/app/src/main/res/drawable-mdpi/images_promotion.png
Comment (harshgarg18) [inline example/android/app/src/main/res/drawable-mdpi/images_promotion.png:1]: why do we have duplication of images between this and next one?

Comment (harshgarg18) [reply example/android/app/src/main/res/drawable-mdpi/images_promotion.png:1]: can you remove the one which is not needed?

File: example/index.js [hunk 7]
```diff
@@ -1,5 +1,17 @@
...
 
 AppRegistry.registerComponent(appName, () => App);
+
+AppRegistry.registerComponent(
+  HyperSdkReact.JuspayHeaderAttached,
+  () => HelloWorld
+);
+AppRegistry.registerComponent(
+  HyperSdkReact.JuspayHeader,
+  () => TopViewAttached
```
Comment (harshgarg18) [inline example/index.js:16]: TopViewAttached is not supposed to be for JuspayHeaderAttached?

File: ios/HyperSdkReact.mm [hunk 8]
```diff
@@ -39,7 +162,10 @@ + (BOOL)requiresMainQueueSetup{
 
 - (NSDictionary *)constantsToExport
 {
- return @{ HYPER_EVENT: HYPER_EVENT };
+ return @{ HYPER_EVENT: HYPER_EVENT
+         , JUSPAY_HEADER : JUSPAY_HEADER
+         , JUSPAY_HEADER_ATTACHED : JUSPAY_HEADER_ATTACHED
+         };
```
Comment (harshgarg18) [inline ios/HyperSdkReact.mm:168]: Same here, there should be JUSPAY_FOOTER and JUSPAY_FOOTER_ATTACHED too

File: src/index.tsx [hunk 9]
```diff
@@ -49,6 +52,9 @@ type HyperSdkReactType = {
   isInitialised(): Promise<boolean>;
   updateBaseViewController(): void;
   openPaymentPage(data: string): void;
+  updateMerchantViewHeight(tag: string, height: number): void;
+  JuspayHeader: string;
+  JuspayHeaderAttached: string;
 };
```
Comment (harshgarg18) [inline src/index.tsx:58]: footer, footer attached

**Review by harshgarg18 — CHANGES_REQUESTED** [review]
Also, Can you get iOS code reviewed by @Subhash703 or @balaganesh-juspay ?

File: ios/HyperSdkReact.mm [hunk 10]
```diff
@@ -154,6 +282,12 @@ -(void)stopObserving {
     }
 }
 
+RCT_EXPORT_METHOD(updateMerchantViewHeight: (NSString * _Nonnull) tag height: (NSNumber * _Nonnull) h) {
```
Comment (balaganesh-juspay) [inline ios/HyperSdkReact.mm:285]: @Georgepadannamackal When do they need to call this function? And will merchant be able to get the exact height of the view?

**Review by balaganesh-juspay — APPROVED** [review]

**Review by harshgarg18 — APPROVED** [review]

**Pull request merged**